storage_service/locations/models/space.py

```diff
@@ -513,7 +513,7 @@ def create_rsync_directory(self, destination_path, user, host):
         for directory in directories:
             path = os.path.join(os.path.dirname(directory), '')
             path = "{}@{}:{}".format(user, host, utils.coerce_str(path))
-            cmd = ['rsync', '-vv', '--protect-args', '--chmod=ugo+rw', '--recursive', temp_dir, path]
+            cmd = ['rsync', '-vv', '--protect-args', '--chmod=ugo+rwx', '--recursive', temp_dir, path]
```

Comment:
@remileduc I like the idea of using ssh and `mkdir -p` to create the rsync directory. However, the third line of the snippet above has an error and will include the string `'path'` instead of the value of the `path` var. It should be something like `cmd = ['ssh', '{}@{}'.format(user, host), '"mkdir -p --mode=a+rwx {}"'.format(path)]`.

Also, in the snippet above `--mode=a+rwx` is being used while different permissions (`--chmod=ug=rwx,o=rx`) are being used in the current state of the PR.

Finally, are you sure that you want to call `path = os.path.dirname(destination_path)`? If `destination_path` has no trailing slash, then this will return the parent directory, which is not what you want, I think. If we want to always have a trailing slash we should continue to join on the empty string:

    >>> import os
    >>> os.path.dirname('a/b/c/')
    'a/b/c'
    >>> os.path.dirname('a/b/c')
    'a/b'
    >>> os.path.join('a/b/c/', '')
    'a/b/c/'
    >>> os.path.join('a/b/c', '')
    'a/b/c/'

Reply:
You're totally right! I did the snippet above very quickly to show the main idea...

Should I update the PR with the `mkdir` version?

Reply:
I think so. Your strategy is a lot cleaner than the current one which needlessly makes a separate rsync call for each ancestor directory.